egg/core/interaction.py

```diff
@@ -67,18 +77,37 @@ class Interaction:
 
     @property
     def size(self):
-        interaction_fields = [
+        for t in self.interaction_fields():
+            if t is not None and isinstance(t, torch.Tensor):
+                return t.size(0)
+        raise RuntimeError("Cannot determine interaction log size; it is empty.")
```

Comment:
This looks a little sketchy.  What if the only available `t` is a `aux` or something else?
Wouldn't it be better if we add some sort of size parameter when the interaction is created?

Reply:
aux can have different size from tensors that are fields of Interaction.
I think that at the moment if we have an aux-only Interaction it would crash only if you're running a distributed training https://github.com/facebookresearch/EGG/blob/56233d3c4545fbf0b8c8c88b3b3221107794c35e/egg/core/interaction.py#L191-L232

feel free to open an issue if you like

Reply:
Still, the purpose of `t.size(0)` is to have the batch_size value. Wouldn't it be better to pass it directly to the Iteration?

Reply:
mmh, I think you might be right actually. Let me think about it and will get back to it as soon as I can

Reply:
Hey @nicofirst1, I was thinking about it and wondered how would you know the size of an interaction beforehand (especially when storing them probabilistically)? I still think having a support method for it would be necessary.